### PR TITLE
Nyaa: Finish Todos

### PR DESCRIPTION
**Pull request by XVicarious**
### Motivation for changes:
- The nyaa plugin wasn't finished

### Detailed changes:
- Added the remaining missing nyaa categories
- Added seeders, leechers, info hash, and content size.

### Log example of accepted entry:
```
2018-11-05 20:26 VERBOSE  urlrewrite_search test_rtorrent   Searching `Kin`iro Mosaic` from nyaa
2018-11-05 20:26 DEBUG    nyaa          test_rtorrent   requesting: https://www.nyaa.si/?page=rss&q=Kin%60iro%20Mosaic&c=1_2&f=0
2018-11-05 20:26 DEBUG    nyaa          test_rtorrent   Search result not 200 (OK), received 301
2018-11-05 20:26 TRACE    entry         test_rtorrent   ENTRY SET: title = 'Patches for [VCB-Studio] Hello!! Kin-iro Mosaic / ハロー!!きんいろモザイク 10bit 1080p HEVC BDRip
[Fin]'
2018-11-05 20:26 TRACE    entry         test_rtorrent   ENTRY SET: original_url = 'https://nyaa.si/download/771798.torrent'
2018-11-05 20:26 TRACE    entry         test_rtorrent   ENTRY SET: url = 'https://nyaa.si/download/771798.torrent'
2018-11-05 20:26 TRACE    entry         test_rtorrent   ENTRY SET: torrent_seeds = 0
2018-11-05 20:26 TRACE    entry         test_rtorrent   ENTRY SET: torrent_leeches = 0
2018-11-05 20:26 TRACE    entry         test_rtorrent   ENTRY SET: torrent_info_hash = 'be2328a6f4b1dfd121890f84015003a04af2f249'
2018-11-05 20:26 TRACE    entry         test_rtorrent   ENTRY SET: search_sort = 0
2018-11-05 20:26 TRACE    entry         test_rtorrent   ENTRY SET: content_size = 3174
2018-11-05 20:26 TRACE    entry         test_rtorrent   ENTRY SET: title = '[Ironanjuu] Hello!! Kiniro Mosaic - 08 [720p][9E37FCCE].mkv'
2018-11-05 20:26 TRACE    entry         test_rtorrent   ENTRY SET: original_url = 'https://nyaa.si/download/694422.torrent'
2018-11-05 20:26 TRACE    entry         test_rtorrent   ENTRY SET: url = 'https://nyaa.si/download/694422.torrent'
2018-11-05 20:26 TRACE    entry         test_rtorrent   ENTRY SET: torrent_seeds = 0
2018-11-05 20:26 TRACE    entry         test_rtorrent   ENTRY SET: torrent_leeches = 0
2018-11-05 20:26 TRACE    entry         test_rtorrent   ENTRY SET: torrent_info_hash = '7c79a7140d7a2a6e4f0641008d13c5f21e1eb7f8'
2018-11-05 20:26 TRACE    entry         test_rtorrent   ENTRY SET: search_sort = 0
2018-11-05 20:26 TRACE    entry         test_rtorrent   ENTRY SET: content_size = 138
2018-11-05 20:26 TRACE    entry         test_rtorrent   ENTRY SET: title = '[Ironanjuu] Hello!! Kiniro Mosaic - 07 [720p] [60137347].mkv'
2018-11-05 20:26 TRACE    entry         test_rtorrent   ENTRY SET: original_url = 'https://nyaa.si/download/691857.torrent'
2018-11-05 20:26 TRACE    entry         test_rtorrent   ENTRY SET: url = 'https://nyaa.si/download/691857.torrent'
2018-11-05 20:26 TRACE    entry         test_rtorrent   ENTRY SET: torrent_seeds = 0
2018-11-05 20:26 TRACE    entry         test_rtorrent   ENTRY SET: torrent_leeches = 0
2018-11-05 20:26 TRACE    entry         test_rtorrent   ENTRY SET: torrent_info_hash = 'd0e7a6462700bf9c0d614ecf86a96488ec161994'
2018-11-05 20:26 TRACE    entry         test_rtorrent   ENTRY SET: search_sort = 0
2018-11-05 20:26 TRACE    entry         test_rtorrent   ENTRY SET: content_size = 154
2018-11-05 20:26 TRACE    entry         test_rtorrent   ENTRY SET: title = '[Ironanjuu] Hello!! Kiniro Mosaic - 06 [720p] [25DCB725].mkv'
2018-11-05 20:26 TRACE    entry         test_rtorrent   ENTRY SET: original_url = 'https://nyaa.si/download/689179.torrent'
2018-11-05 20:26 TRACE    entry         test_rtorrent   ENTRY SET: url = 'https://nyaa.si/download/689179.torrent'
2018-11-05 20:26 TRACE    entry         test_rtorrent   ENTRY SET: torrent_seeds = 0
2018-11-05 20:26 TRACE    entry         test_rtorrent   ENTRY SET: torrent_leeches = 0
2018-11-05 20:26 TRACE    entry         test_rtorrent   ENTRY SET: torrent_info_hash = '5d120053433bc3b440a18eedc2df2e919a682ca9'
2018-11-05 20:26 TRACE    entry         test_rtorrent   ENTRY SET: search_sort = 0
2018-11-05 20:26 TRACE    entry         test_rtorrent   ENTRY SET: content_size = 124
2018-11-05 20:26 TRACE    entry         test_rtorrent   ENTRY SET: title = 'Kin-iro Mosaic (BD 720p) [Kirakira]'
2018-11-05 20:26 TRACE    entry         test_rtorrent   ENTRY SET: original_url = 'https://nyaa.si/download/601824.torrent'
2018-11-05 20:26 TRACE    entry         test_rtorrent   ENTRY SET: url = 'https://nyaa.si/download/601824.torrent'
2018-11-05 20:26 TRACE    entry         test_rtorrent   ENTRY SET: torrent_seeds = 2
2018-11-05 20:26 TRACE    entry         test_rtorrent   ENTRY SET: torrent_leeches = 0
2018-11-05 20:26 TRACE    entry         test_rtorrent   ENTRY SET: torrent_info_hash = '4bc2cc2fabb154d0e7808b776e36c4fb80fe8e95'
2018-11-05 20:26 TRACE    entry         test_rtorrent   ENTRY SET: search_sort = 4
2018-11-05 20:26 TRACE    entry         test_rtorrent   ENTRY SET: content_size = 6451
```

### Additional Details:
For what it is worth, I used this fixed version of the plugin to search for and restore all of the torrents my dog deleted out of ruTorrent last week when he sat on the delete key. Of course I didn't think to collect these logs.

### Todo:
- Let me know how you feel about the shortened names of the categories.
- Size calculation could be better, but it appears to be working properly

### CODE REVIEW

File: flexget/plugins/sites/nyaa.py [hunk 1]
```diff
@@ -68,11 +93,21 @@ def search(self, task, entry, config):
...
-                # entry['torrent_leeches'] = int(item.leechs)
-                # entry['search_sort'] = torrent_availability(entry['torrent_seeds'], entry['torrent_leeches'])
-                # entry['content_size'] = int(item.size) / 1024 / 1024
+                entry['torrent_seeds'] = int(item.nyaa_seeders)
+                entry['torrent_leeches'] = int(item.nyaa_leechers)
+                entry['torrent_info_hash'] = item.nyaa_infohash
+                entry['search_sort'] = torrent_availability(entry['torrent_seeds'], entry['torrent_leeches'])
+                nyaa_size_split = item.nyaa_size.split(' ')
+                if len(nyaa_size_split):
+                    size_multiplier = float(1.0 / 1024.0**2)  # Assume it is in bytes
```
Comment (cvium) [inline flexget/plugins/sites/nyaa.py]: https://github.com/Flexget/Flexget/blob/develop/flexget/utils/tools.py#L442

Comment (XVicarious) [reply flexget/plugins/sites/nyaa.py]: Oh! That's neat. Let's use that.

**Comment by cvium** [comment]
Is this ready then?

**Comment by XVicarious** [comment]
Yep

**Pull request merged**